packages/components/psammead-social-embed/src/Canonical/index.jsx

```diff
@@ -56,19 +56,16 @@ export const providers = {
   },
 };
 
-const CanonicalEmbed = ({ provider, oEmbed }) => (
-  <>
-    {providers[provider].script && (
-      <Helmet>
-        <script async src={providers[provider].script} />
-      </Helmet>
-    )}
+const CanonicalEmbed = ({ provider, oEmbed }) => {
+  useScript(providers[provider].script);
```

Comment:
Instead of a custom hook could we not use helmet here to add the script for us and just wrap it in an `onClient()` check?

Reply:
Hey Chris, Helmet within a conditional might work. I opted for the useEffect approach, because we can [guarantee it won't run on the server](https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85) - I've seen threads about `window` checks failing in test environments (because of JSDOM) and even on the server, so I'm never sure how much it can be trusted. Also useEffect allows us to be explicit about how we clean-up/remove these script elements on unmount.

Reply:
So is it worth using Helmet within the use effect as to not implement a second way of adding scripts to the head, after all this is what Helmet is designed to do.

Reply:
If we use Helmet within a hook, how do you suggest we go about cleaning-up/removing the script on unmount?

Reply:
After chatting with Simon about this It seems like the current approach here is the best fit as we cannot find an easy way to clean up in the useEffect when using Helmet.

Happy to leave this as is.